pipelines/cellranger/cellranger.wdl

```diff
@@ -0,0 +1,92 @@
+        sample_id: "Name of sample to run CellRanger count on"
+        fastqs: "Array of fastq files for running decoupled"
+        reference_name: "Reference name for count"
+        transcriptome_tar_gz: "Cellranger transcriptome reference"
+        expect_cells: "Expected number of recovered cells (defaults to 3000)"
+        disk_space: "Amount of disk space (GB) to allocate to the VM"
+    }
+
+    call CellRanger {
+        input:
+        sample_id = sample_id,
+        fastqs = fastqs,
+        reference = reference_name,
+        transcriptome_tar_gz = transcriptome_tar_gz,
+        expect_cells = expect_cells,
+        disk_space = disk_space
+   }
+}
+
+task CellRanger {
+    String sample_id
+    Array[File] fastqs
+    String reference
+    File transcriptome_tar_gz
+    Int? expect_cells
+    String disk_space
+
+    command {
+        set -e
+        mkdir transcriptome_dir
+        tar xf ${transcriptome_tar_gz} -C transcriptome_dir --strip-components 1
+        ln -s /usr/bin/python3 python
+        export PATH=$PATH:.
+        python <<CODE
+        import os
+        from subprocess import call
+        dirs = dict()
+        for f in ["${sep='","' fastqs}"]:
+            dirs.setdefault(os.path.dirname(f), True)
+        expect_cells = '${expect_cells}'
+        call_args = list()
+        call_args.append('cellranger')
+        call_args.append('count')
+        call_args.append('--jobmode=local')
+        call_args.append('--transcriptome=transcriptome_dir')
+        call_args.append('--id=' + '${sample_id}')
+        call_args.append('--fastqs=' + ','.join(list(dirs.keys())))
+        call_args.append('--nosecondary')
+        call_args.append('--disable-ui')
+        if expect_cells is not '':
+            call_args.append('--expect-cells=' + str(expect_cells))
+        call(call_args)
+        CODE
+    }
+
+    output {
+        File qc = "${sample_id}/outs/metrics_summary.csv"
+        File sorted_bam = "${sample_id}/outs/possorted_genome_bam.bam"
+        File sorted_bam_index = "${sample_id}/outs/possorted_genome_bam.bam.bai"
+        File barcodes = "${sample_id}/outs/filtered_gene_bc_matrices/${reference}/barcodes.tsv"
+        File genes = "${sample_id}/outs/filtered_gene_bc_matrices/${reference}/genes.tsv"
+        File matrix = "${sample_id}/outs/filtered_gene_bc_matrices/${reference}/matrix.mtx"
+        File filtered_gene_h5 = "${sample_id}/outs/filtered_gene_bc_matrices_h5.h5"
+        File raw_gene_h5 = "${sample_id}/outs/raw_gene_bc_matrices_h5.h5"
+        File raw_barcodes = "${sample_id}/outs/raw_gene_bc_matrices/${reference}/barcodes.tsv"
+        File raw_genes = "${sample_id}/outs/raw_gene_bc_matrices/${reference}/genes.tsv"
+        File raw_matrix = "${sample_id}/outs/raw_gene_bc_matrices/${reference}/matrix.mtx"
+        File mol_info_h5 = "${sample_id}/outs/molecule_info.h5"
+    }
+
+    runtime {
+        docker: "quay.io/humancellatlas/secondary-analysis-cellranger"
+        memory: "416 GB"
```

Comment:
Maybe we should expose these as parameters with the existing values as defaults. We have small test datasets that could run on a much smaller machine.